src/intercooler.js

```diff
@@ -1732,7 +1732,9 @@ var Intercooler = Intercooler || (function() {
         if (historyData) {
           processICResponse(historyData["content"], getTargetForHistory($('body')), true);
           if (historyData["yOffset"]) {
-            window.scrollTo(0, historyData["yOffset"])
+             setTimeout(function () {
+                window.scrollTo(0, historyData["yOffset"]);
+             }, 40);
```

Comment:
Good catch! I'm just a little worried about the hardcoded delay here, especially because the delay in `processICResponse` is actually variable. I wonder if the scrolling can be pushed down inside `processICResponse` instead of adding a separate `setTimeout` here.

Looking at `doSwap` inside of `processICResponse`, I see that there is already some scrolling logic at the end that is already controlled by a history-specific `forHistory` parameter. Currently that parameter is `true` for the two calls inside this function, and `false` for all other calls. What if we leave the other calls as passing `false` for that parameter, so they remain unaffected by this change, but actually pass `historyData["yOffset"]` itself instead of `true` here?

Then over in `doSwap`, the `if (forHistory != true) {...}` block would become something like `if (forHistory === false) {...} else if (typeof forHistory === "number") {...}`. And of course, `forHistory` should probably be renamed to `yOffsetForHistory`...

What do you think?